model/firewall.rb

```diff
@@ -17,7 +17,29 @@ def insert_firewall_rule(cidr, port_range)
       port_range: port_range
     )
 
-    private_subnet&.incr_update_firewall_rules
+    private_subnets.map(&:incr_update_firewall_rules)
     fwr
   end
+
+  def destroy
+    DB.transaction do
+      private_subnets.map(&:incr_update_firewall_rules)
+      FirewallsPrivateSubnets.where(firewall_id: id).all.each(&:destroy)
+      super
+    end
+  end
+
+  def associate_with_private_subnet(private_subnet, apply_firewalls: true)
+    add_private_subnet(private_subnet)
+    private_subnet.incr_update_firewall_rules if apply_firewalls
+  end
+
+  def disassociate_from_private_subnet(private_subnet, apply_firewalls: true)
+    FirewallsPrivateSubnets.where(
```

Comment:
nit: is there sequel function to do that?

Reply:
`remove_all_private_subnets`

https://sequel.jeremyevans.net/rdoc/classes/Sequel/Model/Associations/ClassMethods.html

Reply:
yes there is but it calls `delete` which we do not allow. We need this to be `destroy`.